### PR TITLE
Registration controller, Validations, User details service

### PR DESCRIPTION
**Pull request by p1q**

### CODE REVIEW

File: src/main/java/library/spring/controller/InjectDataController.java [hunk 1]
```diff
@@ -2,41 +2,81 @@
...
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 
 @Controller
 @RequestMapping("/inject")
+@Transactional
```
Comment (Sofasmile) [inline src/main/java/library/spring/controller/InjectDataController.java]: do we need transactional here?

File: src/main/java/library/spring/controller/InjectDataController.java [hunk 2]
```diff
@@ -2,41 +2,81 @@
...
+        User user3 = new User("Sameer", "Singh", "sameer.singh@example.com", "sameer", "1");
+        User user4 = new User("Paul", "Smith", "paul.smith@example.com", "paul", "1");
+        User user5 = new User("Victor", "Karah", "victor.karah@gmail.com", "victor", "1");
+        sessionFactory.getCurrentSession().save(user1);
+        sessionFactory.getCurrentSession().save(user2);
+        sessionFactory.getCurrentSession().save(user3);
+        sessionFactory.getCurrentSession().save(user4);
+        sessionFactory.getCurrentSession().save(user5);
+
+        String queryString = "INSERT INTO users_roles (user_id, role_id) VALUES (3, 2);";
```
Comment (Sofasmile) [inline src/main/java/library/spring/controller/InjectDataController.java]: what is that? in controller?

File: src/main/java/library/spring/dao/impl/RoleDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public void deleteRole(Role role) {
+        sessionFactory.getCurrentSession().delete(role);
+    }
+
+    @Override
+    public Optional<Role> getRole(Long roleId) {
+        TypedQuery<Role> query = sessionFactory.getCurrentSession()
```
Comment (Sofasmile) [inline src/main/java/library/spring/dao/impl/RoleDaoImpl.java]: we can use method get() here

File: src/main/java/library/spring/security/PrintInfoSuccessHandler.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+public class PrintInfoSuccessHandler implements AuthenticationSuccessHandler {
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest request,
+            HttpServletResponse response, Authentication authentication)
+            throws IOException, ServletException {
+        String username = authentication.getName();
+
+        Collection<? extends GrantedAuthority> authorities = authentication.getAuthorities();
+        for (GrantedAuthority authority : authorities) {
+            System.out.println(authority);
```
Comment (Sofasmile) [inline src/main/java/library/spring/security/PrintInfoSuccessHandler.java]: sout? better use logger

File: src/main/java/library/spring/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
+    private RoleDao roleDao;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
     @Transactional
     @Override
-    public void addUser(User user) {
+    public void addUser(UserDto userDto) {
+        User user = new User(userDto.getName(), userDto.getSurname(),
```
Comment (Sofasmile) [inline src/main/java/library/spring/service/impl/UserServiceImpl.java]: suggestion: better create separate method for mapping

File: src/main/java/library/spring/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -1,22 +1,41 @@
...
+                userDto.getEmail(), userDto.getLogin(), userDto.getPassword());
+        user.setPassword(passwordEncoder.encode(userDto.getPassword()));
+        Role role = roleDao.getRoleByName(ROLE_USER).orElseThrow(()
```
Comment (Sofasmile) [inline src/main/java/library/spring/service/impl/UserServiceImpl.java]: better move this line in registration controller

File: src/main/java/library/spring/dao/impl/RoleDaoImpl.java (hunk 3 above)
Comment (p1q) [reply src/main/java/library/spring/dao/impl/RoleDaoImpl.java]: Which method get()?

File: src/main/java/library/spring/controller/InjectDataController.java (hunk 1 above)
Comment (p1q) [reply src/main/java/library/spring/controller/InjectDataController.java]: We need it to inject initial data.

File: src/main/java/library/spring/dao/impl/RoleDaoImpl.java (hunk 3 above)
Comment (Sofasmile) [reply src/main/java/library/spring/dao/impl/RoleDaoImpl.java]: приклад:
session = HibernateUtil.getSessionFactory().openSession();
            user =  (User) session.get(User.class, user_id);

File: src/main/java/library/spring/controller/InjectDataController.java [hunk 7]
```diff
@@ -27,10 +26,10 @@
 @EnableTransactionManagement
 public class InjectDataController {
     @Autowired
-    private SessionFactory sessionFactory;
+    private RentDao rentDao;
```
Comment (Sofasmile) [inline src/main/java/library/spring/controller/InjectDataController.java]: why dao?

File: src/main/java/library/spring/dao/impl/RoleDaoImpl.java (hunk 3 above)
Comment (p1q) [reply src/main/java/library/spring/dao/impl/RoleDaoImpl.java]: Okay, done, thanks.

File: src/main/java/library/spring/controller/InjectDataController.java (hunk 1 above)
Comment (Sofasmile) [reply src/main/java/library/spring/controller/InjectDataController.java]: yes, but you can use methods from service

File: src/main/java/library/spring/controller/InjectDataController.java (hunk 7 above)
Comment (p1q) [reply src/main/java/library/spring/controller/InjectDataController.java]: Fixed

File: src/main/java/library/spring/controller/InjectDataController.java (hunk 1 above)
Comment (p1q) [reply src/main/java/library/spring/controller/InjectDataController.java]: Agree, already done

**Pull request merged**